Add support for Cilium network policies (#363)
DataDog/datadog-operator: docs/configuration.md

| agent.log.podLogsPath | Allows log collection from pod log path. Defaults to `/var/log/pods`. |
| agent.log.tempStoragePath | This path (always mounted from the host) is used by Datadog Agent to store information about processed log files. If the Datadog Agent is restarted, it starts tailing the log files immediately. Default to `/var/lib/datadog-agent/logs` |
| agent.networkPolicy.create | If true, create a NetworkPolicy for the current agent. |

> | agent.networkPolicy.dnsSelectorEndpoints |  |

Reviewer: Needs a description.

> | agent.networkPolicy.flavor | Flavor of the network policy to use. Can be `kubernetes` or `cilium`. |

Reviewer: ```suggestion
| agent.networkPolicy.flavor | Which network policy to use. Can be `kubernetes` or `cilium`. |
```

| clusterAgent.keepAnnotations | KeepAnnotations allows the specification of annotations not managed by the Operator that will be kept on ClusterAgent Deployment. All annotations containing 'datadoghq.com' are always included. This field uses glob syntax. |
| clusterAgent.keepLabels | KeepLabels allows the specification of labels not managed by the Operator that will be kept on ClusterAgent Deployment. All labels containing 'datadoghq.com' are always included. This field uses glob syntax. |
| clusterAgent.networkPolicy.create | If true, create a NetworkPolicy for the current agent. |
| clusterAgent.networkPolicy.dnsSelectorEndpoints |  |

> | clusterAgent.networkPolicy.flavor | Flavor of the network policy to use. Can be `kubernetes` or `cilium`. |

Reviewer: Same comments as above

| clusterChecksRunner.image.pullSecrets | It is possible to specify docker registry credentials. See https://kubernetes.io/docs/concepts/containers/images/#specifying-imagepullsecrets-on-a-pod |
| clusterChecksRunner.image.tag | Define the image version to use: To be used if the Name field does not correspond to a full image string. |
| clusterChecksRunner.networkPolicy.create | If true, create a NetworkPolicy for the current agent. |
| clusterChecksRunner.networkPolicy.dnsSelectorEndpoints |  |

> | clusterChecksRunner.networkPolicy.flavor | Flavor of the network policy to use. Can be `kubernetes` or `cilium`. |

Reviewer: same comments as above.
